src/lib/hooks/useSwapDefaults.ts

```diff
@@ -0,0 +1,57 @@
+import { useUpdateAtom } from 'jotai/utils'
+import { DefaultAddress } from 'lib/components/Swap'
+import { useSwapAmount, useSwapCurrency } from 'lib/hooks/swap'
+import useActiveWeb3React from 'lib/hooks/useActiveWeb3React'
+import { useToken } from 'lib/hooks/useCurrency'
+import { Field, swapDefaultsAtom } from 'lib/state/swap'
+import { useEffect } from 'react'
+
+export default function useSwapDefaults(
+  defaultInputAddress: DefaultAddress | undefined,
+  defaultInputAmount: string | undefined,
+  defaultOutputAddress: DefaultAddress | undefined,
+  defaultOutputAmount: string | undefined
+) {
+  const setDefaults = useUpdateAtom(swapDefaultsAtom)
+  const [inputCurrency, updateSwapInputCurrency] = useSwapCurrency(Field.INPUT)
+  const [outputCurrency, updateSwapOutputCurrency] = useSwapCurrency(Field.OUTPUT)
+  const [inputAmount, updateSwapInputAmount] = useSwapAmount(Field.INPUT)
+  const [outputAmount, updateSwapOutputAmount] = useSwapAmount(Field.OUTPUT)
+  const { chainId } = useActiveWeb3React()
+
+  const inputAddress =
+    typeof defaultInputAddress === 'string'
+      ? defaultInputAddress
+      : defaultInputAddress && chainId
+      ? defaultInputAddress[chainId]
+      : undefined
+  const defaultInputToken = useToken(inputAddress)
+
+  const outputAddress =
+    typeof defaultOutputAddress === 'string'
+      ? defaultOutputAddress
+      : defaultOutputAddress && chainId
+      ? defaultOutputAddress[chainId]
+      : undefined
+  const defaultOutputToken = useToken(outputAddress)
+
+  useEffect(() => {
+    // alter default values if they have changed
```

Comment:
I don't think defaults should be able to change - they should just be set on first render. wdyt?

Reply:
I figured someone's going to run into issues expecting it to work this way one day, and it's not hard to enable it, so why not. Not married to it tho.